static/src/javascripts/bootstraps/enhanced/media/main.js

```diff
@@ -179,32 +179,36 @@ define([
         var mediaType = el.tagName.toLowerCase(),
             $el = bonzo(el).addClass('vjs vjs-tech-' + videojs.options.techOrder[0]),
             mediaId = $el.attr('data-media-id'),
-            blockVideoAds = $el.attr('data-block-video-ads') === 'true',
             showEndSlate = $el.attr('data-show-end-slate') === 'true',
             endSlateUri = $el.attr('data-end-slate'),
             embedPath = $el.attr('data-embed-path'),
             // we need to look up the embedPath for main media videos
             canonicalUrl = $el.attr('data-canonical-url') || (embedPath ? '/' + embedPath : null),
             techPriority = techOrder(el),
-            withPreroll = shouldPreroll && !blockVideoAds,
             player,
             mouseMoveIdle,
             playerSetupComplete,
+            withPreroll,
+            blockVideoAds,
             isPlayerExpired;
 
-        isPlayerExpired = new Promise(function(resolve) {
+        var videoInfo = new Promise(function(resolve) {
             // We only have the canonical URL in videos embedded in articles / main media.
             if (!canonicalUrl) {
                 resolve(false);
             } else {
                 ajax({
                     url: canonicalUrl + '/info.json'
                 }).then(function(videoInfo) {
-                    resolve(videoInfo.expired);
+                    resolve(videoInfo);
                 });
             }
         });
 
+        isPlayerExpired = videoInfo.expired;
+        blockVideoAds = videoInfo.shouldHideAdverts;
+        withPreroll = shouldPreroll && !blockVideoAds;
+
```

Comment:
^ These aren't available at this point - they are part of the promise, so you would need to refer to them in the resolution.